src/include/etherfabric/checksum.h

```diff
@@ -97,6 +97,32 @@ extern uint32_t
 ef_udp_checksum_ipx(int af, const void* ipx, const struct udphdr* udp,
                     const struct iovec* iov, int iovlen);
 
+/*! \brief Calculate the checksum for a UDP packet
+**
+** \param af            The address family of the IP header for the packet.
+** \param ipx           The IP header for the packet (struct iphdr or ipv6hdr).
+** \param udp           The UDP header for the packet.
+** \param payload       UDP payload.
+** \param payload_len   Length of the payload.
+**
+** \return The checksum of the UDP packet.
+**
+** Calculate the checksum for a UDP packet.  The UDP header must be populated
+** (with the exception of the checksum field itself, which is ignored) before
+** calling this function.
+*/
+ef_vi_inline uint32_t
```

Comment:
Bit weird that you're returning u32 rather than u16, but it's consistent with the existing stuff, so OK